src/sentry/static/sentry/app/views/performance/trends/content.tsx

```diff
@@ -28,7 +31,42 @@ type State = {
 };
 
 class TrendsContent extends React.Component<Props, State> {
+  constructor(props) {
+    super(props);
+
+    const {location} = props;
+
+    const queryString = decodeScalar(location.query.query) || '';
+
+    const conditions = tokenizeSearch(queryString);
+
+    if (!queryString) {
+      conditions.setTag('count()', ['>1000']);
+      conditions.setTag('transaction.duration', ['>0']);
+    }
+    if (!conditions.hasTags('count()')) {
+      conditions.setTag('count()', ['>1000']);
+    }
+    if (!conditions.hasTags('transaction.duration')) {
+      conditions.setTag('transaction.duration', ['>0']);
+    }
+
+    const query = stringifyQueryObject(conditions);
+    props.eventView.query = query;
+
+    browserHistory.push({
+      pathname: location.pathname,
+      query: {
+        ...location.query,
+        cursor: undefined,
+        query: String(query).trim() || undefined,
+        view: FilterViews.TRENDS,
+      },
```

Comment:
I'm guessing you want to have the default query string values in the URL bar? Redirecting from a constructor is unconventional and might not stop the first render.

Reply:
it won't stop the first render, but it does push correctly, that's why I'm doing `props.eventView.query = query;` to make sure the eventview also has these appended tags.

Reply:
Fixed to use intermediate component like we talked about 👍